### PR TITLE
Enable AESGCM encryption of secrets in etcd by default.

### PR DESCRIPTION
**Pull request by immutableT**
**What this PR does / why we need it**:
Enable encryption of secrets in etcd via AESGCM transform (as described here https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) during kube-up.sh build of a cluster.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -194,6 +194,24 @@ if [[ ${ENABLE_METADATA_CONCEALMENT:-} == "true" ]]; then
   PROVIDER_VARS="${PROVIDER_VARS:-} ENABLE_METADATA_CONCEALMENT METADATA_CONCEALMENT_NO_FIREWALL"
 fi
 
+# Enable AESGCM encryption of secrets by default.
+ENCRYPTION_PROVIDER_CONFIG="${ENCRYPTION_PROVIDER_CONFIG:-}"
+if [[ -z "${ENCRYPTION_PROVIDER_CONFIG}" ]]; then
+    ENCRYPTION_PROVIDER_CONFIG=$(cat << EOM | base64 -w 0
```
Comment (roberthbailey) [inline cluster/gce/config-default.sh]: Have you tested this on a mac? See https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L1026 where I had to work around apple's base64 with `tr`.

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -818,6 +818,7 @@ ETCD_CA_KEY: $(yaml-quote ${ETCD_CA_KEY_BASE64:-})
 ETCD_CA_CERT: $(yaml-quote ${ETCD_CA_CERT_BASE64:-})
 ETCD_PEER_KEY: $(yaml-quote ${ETCD_PEER_KEY_BASE64:-})
 ETCD_PEER_CERT: $(yaml-quote ${ETCD_PEER_CERT_BASE64:-})
+ENCRYPTION_PROVIDER_CONFIG: $(yaml-quote ${ENCRYPTION_PROVIDER_CONFIG:-})
```
Comment (roberthbailey) [inline cluster/gce/util.sh]: Setting it here will put it into the metadata, but presumably you need to also make a change on the master to apply the new resource? Or is that code already in place?

Comment (immutableT) [reply cluster/gce/util.sh]: Yes, such code already exists:
https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L1654

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (immutableT) [reply cluster/gce/config-default.sh]: Thanks for catching this; I was not aware of -w not being available on Macs.
Replace -w 0 with | tr -d '\r\n'

**Comment by BenTheElder** [comment]
flakes are probably https://github.com/kubernetes/kubernetes/issues/60614

**Comment by BenTheElder** [comment]
pull-kubernetes-unit is pull-kubernetes-integration now for clarity, that job status was in the middle of the migration and I've left it so as not to clobber your green pull-kubernetes-integration run with a broken unit run
/skip
/test pull-kubernetes-integration

**Comment by BenTheElder** [comment]
Hmm, I'm not seeing that integration failure on other PRs and haven't seen it before, but it seems to be quite reliable here 😕

**Comment by immutableT** [comment]
Yes, interesting.
Will need to dig into that test.

**Comment by roberthbailey** [comment]
Please squash your commits.

**Comment by immutableT** [comment]
Done.

**Pull request merged**